### DISCUSSION TITLE
Vaultwarden container exiting for no reason

### DISCUSSION THREAD

**Discussion by rishubn** [dani-garcia/vaultwarden#2794] [Q&A] [answered]
### Subject of the issue
Vaultwarden container exiting for no reason (?)

### Deployment environment

* vaultwarden version: 1.25.2

* Install method: container (podman 3.4.4), amd64 PC

* Other relevant details:

### Steps to reproduce
podman run -d --name vaultwarden -v $PWD/data/vw-data:/data/:Z -e ROCKET_PORT=8080 -p 8080:8080 vaultwarden/server:latest
### Expected behaviour
The container stays alive until killed by user or shutdown

### Actual behaviour
Container runs for several minutes then shuts down for seemingly no reason - no reason is given in the log file

### Troubleshooting data
```
[2022-10-04 08:57:33.528][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2022-10-04 08:57:33.567][request][INFO] GET /icons/xilinx.com/icon.png
[2022-10-04 08:57:33.568][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2022-10-04 09:00:17.105][vaultwarden][INFO] Exiting vaultwarden! <---- exits after 3 minutes unprompted
[2022-10-04 09:00:17.105][vaultwarden][INFO] Exiting vaultwarden!
[2022-10-04 09:00:17.105][rocket::server][WARN] Received SIGTERM. Shutdown already in progress.
[2022-10-04 09:00:17.106][vaultwarden][INFO] Vaultwarden process exited!
```

Anyway to get more detailed logs?

**Comment by jonas-w** [comment, marked as answer]
```
[2022-10-13 22:26:05.153][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2022-10-13 22:51:44.680][vaultwarden][INFO] Exiting vaultwarden!
[2022-10-13 22:51:44.680][rocket::server][WARN] Received SIGTERM. Requesting shutdown.
[2022-10-13 22:51:44.682][vaultwarden][INFO] Vaultwarden process exited!
```
exactly the same thing happened 2 days ago for me.
I'm using version 1.25.2 (docker, Ubuntu LTS).


Maybe this could be "escalated" to an issue again, as this happened to both of us and maybe has something to do with version 1.25.2 as this never happened before.

**Reply by BlackDex** [reply]
Then i would suggest to update to 1.26.0.

But this isn't something we can debug. And mostly this is done by the host/system for some reason, a SIGTERM is sent from external and not a crash or something.

It could be OOM (Out Of Memory), or any other system resource going beyond some defined metric to kill items. Even updates on the host could have done this.

**Reply by BlackDex** [reply]
Also, to get more detailed logs from Vaultwarden you can set `LOG_LEVEL` to either `debug` or `trace`. But i would suggest to check your host logs first from around the time it happened.

**Reply by jonas-w** [reply]
Ah okay, i misunderstood the "Received SIGTERM" because it was after Vaultwarden's exiting message. I thought the SIGTERM came from vaultwarden itself because it wants to exit.

**Reply by jonas-w** [reply]
I updated to 1.26 but found at least something in the log.
The ID is the vaultwarden container. But this is written to the log every 2-5 minutes, so nothing unusual (i think?) but 20 seconds after that the vaultwarden instance got the SIGTERM.
```
Oct 13 22:51:26 kumquat.03j.de systemd[1]: run-docker-runtime\x2drunc-moby-7c04cb7926efab4493e01322ffa04b576d058c220903d366bc2046978eb0c707-runc.w65XPT.mount: Succeeded.
Oct 13 22:51:26 kumquat.03j.de systemd[72683]: run-docker-runtime\x2drunc-moby-7c04cb7926efab4493e01322ffa04b576d058c220903d366bc2046978eb0c707-runc.w65XPT.mount: Succeeded.
```

**Reply by rishubn** [reply]
@jonas-w I solved the problem with `loginctl enable-linger` - I think systemd was killing the process beforehand

**Reply by jonas-w** [reply]
@rishubn thanks i will try this, if it happens again.

**Reply by jonas-w** [reply]
It just happened again i just realized it happened nearly at the same time.
First time it was 22:51:40 and now it was 22:45:40, i checked my cron jobs and none are executed at that time, will check if something else is triggered at this time.

**Reply by EricDriussi** [reply]
This has happened to me a couple of times but I'm unable to see a pattern or find anything in the logs.
I've got a hunch that systemd might be envolved but I can't find out for certain.
I've also checked`journalctl --list-boots` (maybe it wasn't restarting after maintenance on the server?) but there doesn't seem to be a relation.

Any idea how I can further troubleshoot this? Which logs to check specifically or what to try?

**Reply by EricDriussi** [reply]
Jesus Crist I was running on 1.24 -.-U Prolly gonna fix itself now

**Comment by mattv8** [comment]
Something very similar is happening to me on the latest version (2023-11). Should I open new issue or post my logs here?

**Reply by BlackDex** [reply]
Tried testing already?

**Reply by mattv8** [reply]
Hey I'm sorry for the delayed response. I was able to pin my issue down. It was due to misconfigured SMTP server settings. Once I corrected those, things became much more stable.
